examples/example_ensemble_classification.py

```diff
@@ -0,0 +1,227 @@
+
+import numpy as np
+
+import sklearn.datasets
+import sklearn.model_selection
+from sklearn.utils.multiclass import type_of_target
+from sklearn.metrics import accuracy_score
+
+from autoPyTorch.datasets.tabular_dataset import TabularDataset
+from autoPyTorch.ensemble.ensemble_builder import EnsembleBuilderManager
+from autoPyTorch.pipeline.tabular_classification import TabularClassificationPipeline
+from autoPyTorch.pipeline.components.training.metrics.metrics import accuracy
+from autoPyTorch.utils.backend import create, Backend
+from autoPyTorch.constants import MULTICLASS, TABULAR_CLASSIFICATION
+
+
+def get_data_to_train() -> typing.Tuple[typing.Dict[str, typing.Any]]:
+    """
+    This function returns a fit dictionary that within itself, contains all
+    the information to fit a pipeline
+    """
+
+    # Get the training data for tabular classification
+    # Move to Australian to showcase numerical vs categorical
+    X, y = sklearn.datasets.fetch_openml(data_id=40981, return_X_y=True, as_frame=True)
+    X_train, X_test, y_train, y_test = sklearn.model_selection.train_test_split(
+        X,
+        y,
+        random_state=1,
+        test_size=0.2,
+    )
+
+    train_indices, val_indices = sklearn.model_selection.train_test_split(
+        list(range(X_train.shape[0])),
+        random_state=1,
+        test_size=0.25,
+    )
+
+    output_type = type_of_target(y)
+
+    # Mock the categories
+    categorical_columns = ['A1', 'A4', 'A5', 'A6', 'A8', 'A9', 'A11', 'A12']
+    numerical_columns = ['A2', 'A3', 'A7', 'A10', 'A13', 'A14']
+    categories = [np.unique(X[a]).tolist() for a in categorical_columns]
+
+    # Create a proof of concept pipeline!
+    dataset_properties = {
+        'task_type': 'tabular_classification',
+        'categorical_columns': categorical_columns,
+        'numerical_columns': numerical_columns,
+        'output_type': output_type,
+    }
+    # Fit the pipeline
+    fit_dictionary = {
+        'categorical_columns': categorical_columns,
+        'numerical_columns': numerical_columns,
+        'num_features': X.shape[1],
+        'num_classes': len(np.unique(y)),
+        'is_small_preprocess': True,
+        'categories': categories,
+        'X_train': X_train,
+        'y_train': y_train,
+        'train_indices': train_indices,
+        'val_indices': val_indices,
+        'X_test': X_test,
+        'y_test': y_test,
+        'dataset_properties': dataset_properties,
+        # Training configuration
+        'job_id': 'example_ensemble_1',
+        'working_dir': '/tmp/example_ensemble_1',  # Hopefully generated by backend
+        'device': 'cpu',
+        'runtime': 100,
+        'torch_num_threads': 1,
+        'early_stopping': 20,
+        'use_tensorboard_logger': True,
+        'use_pynisher': False,
+        'memory_limit': 4096,
+        'metrics_during_training': True,
+        'seed': 0,
+        'budget_type': 'epochs',
+        'epochs': 10.0,
+        'id': 0,
```

Comment:
`id` is a Python built-in, so I'd prefer another name (I think autosklearn also uses this)

Reply:
I think this is not needed and I have removed it.